sdks/python/apache_beam/io/gcp/datastore/v1/datastoreio.py

```diff
@@ -313,8 +313,11 @@ class _Mutate(PTransform):
   supported, as the commits are retried when failures occur.
   """
 
-  # Max allowed Datastore write batch size.
+  # Max allowed Datastore writes per batch, and max bytes per batch.
+  # Note that the max bytes per batch is lower than the actual limit enforced
```

Comment:
It does result in more RPCs, but 10MB is a limit, not a target. Most workloads will not hit 5MB either I guess.

The 10MB limit includes the whole RPC, not just the mutations, so some space needs to be left over for other fields in the CommitRequest. That said, halving the limit is very conservative; at the time I was thinking that this went through an external library (datastorehelper) as well, but I see that the commit path does not. I could make it 9MB if you prefer.